src/web/routes/figma/schemas.ts

```diff
@@ -1,89 +1,50 @@
 import type {
 	FigmaOAuth2CallbackQueryParameters,
-	FigmaWebhookEventPayload,
+	FigmaWebhookEventRequest,
 } from './types';
 
-import type {
-	JSONSchemaType,
-	JSONSchemaTypeWithId,
-} from '../../../common/schema-validation';
-import type { FigmaWebhookEventType } from '../../../domain/entities';
-
-const FIGMA_WEBHOOK_EVENT_TYPE_SCHEMA: JSONSchemaType<FigmaWebhookEventType> = {
-	type: 'string',
-	enum: [
-		'PING',
-		'FILE_UPDATE',
-		'FILE_VERSION_UPDATE',
-		'FILE_DELETE',
-		'LIBRARY_PUBLISH',
-		'FILE_COMMENT',
-	],
-};
-
-/**
- * While AJV supports optional fields in schemas, it requires these fields to
- * also be marked as nullable. If you mark an optional field as non-nullable,
- * the schema validation works correctly but the typechecking complains that
- * the field is not marked nullable.
- *
- * This function works around the typechecking by casting the field schema to
- * the appropriate type. This is supposed to be fixed in the next major version
- * of AJV.
- *
- * @see https://github.com/ajv-validator/ajv/issues/1664#issuecomment-873613644
- */
-const optionalNonNullable = <T>(schema: T): T & { nullable: true } =>
-	schema as T & { nullable: true };
+import type { JSONSchemaTypeWithId } from '../../../common/schema-validation';
 
-export const FIGMA_WEBHOOK_EVENT_REQUEST_SCHEMA: JSONSchemaTypeWithId<{
-	body: FigmaWebhookEventPayload;
-}> = {
+export const FIGMA_WEBHOOK_EVENT_REQUEST_SCHEMA = {
```

Comment:
**note:** Improve schema definition by using [JSON Schema Discriminator](https://ajv.js.org/json-schema.html#discriminator) (e.g., `file_key` gets required for the `FILE_UPDATE` event but is still optional for `PING`).